src/stats/src/GaussianLikelihoodFullCovarianceRandomCoefficient.C

```diff
@@ -81,9 +81,19 @@ GaussianLikelihoodFullCovarianceRandomCoefficient<V, M>::lnValue(
   // This is square of 2-norm
   double norm2_squared = modelOutput.sumOfComponents();
 
+  // Get the determinant of the covariance matrix |\Sigma|
+  double deter_cov = this->m_covariance.determinant();
+
+  //sqrt(|\Sigma|)
+  deter_cov = std::sqrt(deter_cov);
+
+  // Set the right hyperparameter coefficient
   // The last element of domainVector is the multiplicative coefficient of the
   // covariance matrix
-  return -0.5 * norm2_squared / (domainVector[domainVector.sizeLocal()-1]);
+  double cov_coeff = domainVector[domainVector.sizeLocal()-1];
+  cov_coeff = std::pow(std::sqrt(cov_coeff), domainVector.sizeGlobal()); 
+
+  return -0.5 * norm2_squared/ (domainVector[domainVector.sizeLocal()-1]) - std::log((cov_coeff*deter_cov));
```

Comment:
1.  `deter_cov` is constant.  Is there a reason this is computed but 2 \pi^{d/2} is not computed?
2.  This is not crucial, but `domainVector.sizeGlobal()` is an integer.  `std::pow` can be computed more efficiently as a `for` loop if you wish.
3.  Taking `std::log` after a `std::sqrt` can be made more efficient by computing `0.5 * std::log`.